prevent-broker-scale-down-if-contains-partition .md

```diff
@@ -0,0 +1,60 @@
+
+## Proposal
+
+This proposal suggest how we can add the check to detect if the broker still contains any partitions and what to do if the broker scale down is not possible.
+
+Checking if the broker contains partition
+The best way to detect if the broker contains any partitions is to query the topics.
+To do this, we will have to make use of the `AdminClient` class.
+We should probably put the logic for this check inside the `scaleDown` method present in the `KafkaReconciler` class.
+The `scaleDown` method gets triggered when we scale down the no. of replicas in the `Kafka CR`.
+So this check will target the broker which are going to be removed ie. the pods with the highest number.
+
+Methods to be created in the process:
+
+- `getTopicNameAndDescription` (Boolean Type) - This method will be placed inside the `scaleDown()` method in `KafkaReconciler` class and will provide us with the Collection of `TopicDescription`.
+- `describeTopics()` and `topicNames()`  - These two methods will provide us with the description and names of all the topics ie first we will get the names of the topics using the `topicNames` method and using these names we will extract the description of the topics.
+- `brokerHasAnyTopics()` - This method will take as input a podId and a `Collection<TopicDescription>`.
+  It will go through all the broker and see if any partitions are assigned to it.
+
+### Flow :
+
+When the replicas are changed inside the Kafka resource, reconciliation will happen.
+During this reconciliation, we move to the `scaleDown()` method present in the `KafkaReconciler` class.
+The `scaleDown()` method contains the complete mechanism on how the broker is scaled down. So there, we will first create our `adminClient` and provide it with the Secrets.
+Then we will call the `checkPartitions()` method which will give us the `Collection<TopicDescription>`. 
+
+The `checkPartition()` method will call the `topicNames()` method to get the set of topic names and then these topic names will be used by the `describeTopics()` methods to get the `Collection<TopicDescription>`.
+Once the `Collection<TopicDescription>` is computed we will move back to the `scaledown()` method and there we will run a loop from `(previous replicas - 1)` to `current replicas` and inside the loop we will call the `brokersHasAnyTopics()` method which will take the `Collection<TopicDescription>` and `podId(index of the loop)` to check if the broker contains any partition and provide us with a boolean result.
+
+
+### What to do if a broker contains partitions?
+If a partition is detected over a broker then we should trigger some log warnings and let the partitions remain the same as they were before.
+For example, if we try to scale down the broker from 5 to 3, but the check detected that the broker to be scaled down contains a partition then we should keep the replicas as 5 only display the appropriate warnings.
```

Comment:
This needs more details on how do you achieve keeping 5 replicas. It sounds easier than it might be.